lib/engines/content_block_manager/app/services/content_block_manager/get_preview_content.rb

```diff
@@ -32,9 +37,21 @@ def frontend_path
       frontend_base_path + content_item["base_path"]
     end
 
-    def html
+    def preview_html
       uri = URI(frontend_path)
-      @html ||= Nokogiri::HTML.parse(Net::HTTP.get(uri))
+      nokogiri_html = Nokogiri::HTML.parse(Net::HTTP.get(uri))
+      replace_existing_content_blocks(nokogiri_html)
+    end
+
+    def replace_existing_content_blocks(nokogiri_html)
+      existing_content_block_spans(nokogiri_html).each do |span|
+        span.replace @content_block_edition.render
```

Comment:
Tiny suggestion - could we memoize the result of `render` here, so we don't have to potentially call it multiple times?